src/aspire/utils/numeric/pyfftw_fft.py

```diff
@@ -0,0 +1,128 @@
+from threading import Lock
+import os
+import pyfftw
+import pyfftw.interfaces.scipy_fftpack as scipy_fft
+
+mutex = Lock()
+
+_cpu_count = os.cpu_count()
+
+
+def _workers(workers):
```

Comment:
Thanks Junchao .. I'm thinking since we're implementing this anyway, might as well put in the 'wrap around' feature for -ve values. I know I said that was too complex, but the user may want this flexibility (i.e. use all but 2 cpus etc.). It's just a couple of extra lines I think. We can steal the implementation from `scipy.fft` (and put in a note that we've used their logic).

Other than that this PR looks fine to me. Thanks.

Reply:
OK. Done in the new commit. Will change PR to develop branch.